src/app/page/page.component.ts

```diff
@@ -593,6 +595,69 @@ export class PageComponent implements OnInit, OnDestroy {
       .subscribe(() => {
         this.onPreviousPage();
       });
+
+    // Go to any page on page event
+    this.awaitPageEvent();
+  }
+
+  private awaitPageEvent(): void {
+    // We listen for page events and dismiss events
+    // We then navigate to the page that has the event
+    const allListenersOnAllPages = this._pageBookSubPages.reduce(
+      (allListeners, page) => {
+        return allListeners.concat(
+          page.listeners.map((listener) => listener.name),
+          page.dismissListeners.map((listener) => listener.name)
+        );
+      },
+      []
+    );
+
+    this.pageService.contentEvent$
+      .pipe(
+        filter((event) =>
+          allListenersOnAllPages.some((allListenersOnPage) =>
+            allListenersOnPage.includes(event)
+          )
+        )
+      )
+      .pipe(takeUntil(this._unsubscribeAll))
+      .subscribe((event) => {
+        this.navigateToPageOnEvent(event);
+      });
+  }
+
+  private navigateToPageOnEvent(event: string): void {
+    let isListener = false;
+    let isDismissListener = false;
+    let pageToNavigateTo = null;
+    this._pageBookSubPages.forEach((page) => {
+      const foundListener = page.listeners.some(
+        (listener) => listener.name === event
+      );
+      const foundDismissListener = page.dismissListeners.some(
+        (listener) => listener.name === event
+      );
+
+      if (foundListener || foundDismissListener) {
+        pageToNavigateTo = page;
+        isListener = foundListener;
+        isDismissListener = foundDismissListener;
+      }
+    });
+
+    if (isListener) {
+      this.router.navigate([
+        this._pageParams.langid,
+        this._pageParams.bookid,
+        pageToNavigateTo.position
+      ]);
+    }
+
+    if (isDismissListener) {
+      // We want to hide the page, for now I've set this to go to the next page.
+      this.onNextPage();
```

Comment:
hmm, I wonder if we actually do have an dismiss listeners defined for any pages (outside of CYOA where it would navigate to the previous page that launched the page).

Reply:
I doubt we would have dismiss listeners outside of CYOA, as they could just call go to the previous or next page.
I'll update it to go to the previous page.

Reply:
Confirmed with @frett that with CYOA, if page 4 navigated to page 20, the dismiss listener should navigate the user back to page 4.
---
I'll leave it to go to the next page. as CYOA isn't in this PR, and we can build the logic for that in a later PR